### ISSUE TITLE
FALSE POSITIVE: NP_ALWAYS_NULL reports inside a contradictory instanceof branch

### ISSUE THREAD

**Issue by chisa22** [spotbugs/spotbugs#4269]
Hi, I found a false positive in SpotBugs 4.10.3 when the instanceof conjunction cannot be true.

**Affected tool**

`SpotBugs 4.10.3`

**Affected checker**

`SpotBugs NP_ALWAYS_NULL`

**Minimal reproducer**

```java
class SpotBugsContradictoryInstanceof {
  void test(Object value) {
    if (value instanceof String && !(value instanceof Object)) {
      Object object = null;
      object.toString();
    }
  }
}
```

**Reproduction command**

```bash
spotbugs -version
javac -d classes spotbugs-np-always-null-fp-contradictory-instanceof.java
spotbugs -textui -effort:max -low -visitors FindNullDeref classes
```

**Current behavior**

SpotBugs reports `NP_ALWAYS_NULL` at line 5.

**Expected behavior**

SpotBugs should not report `NP_ALWAYS_NULL` because the instanceof conjunction cannot be true.
